server/controllers/userController.js

```diff
@@ -235,7 +304,7 @@ const deleteUser = async (req, res) => {
 };
 
 UserRouter.route('/login').post(emailValidator, loginUser);
-UserRouter.route('/profile').post(emailValidator, updateProfile);
+UserRouter.route('/profile').post(upload.single('profileImage'), emailValidator, updateProfile);
```

Comment:
please explain the logic here

Reply:
upload is an instance of Multer configured with that storage engine.(let's say imageUpload)

upload.single('....') which is used as middleware in route to handle a single file upload(profileImage). if there is no image it will simply pass control to the next middleware in the chain(emailValidator ).

I like to know if there is good way to handle it, because we always don't update|upload image. ( another way - create separate post for image and else ?)

Reply:
it's good, no need for the extra post